jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/helper/JKubeDockerfileInterpolator.java

```diff
@@ -13,37 +13,197 @@
  */
 package org.eclipse.jkube.kit.build.service.docker.helper;
 
+import org.apache.commons.lang3.StringUtils;
+
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Properties;
 
 public class JKubeDockerfileInterpolator {
-    private static Map<String, String> delimiters;
-    static {
-        delimiters = new HashMap<>();
-        delimiters.put("@", "@");
-        delimiters.put("${", "}");
-    }
 
     private JKubeDockerfileInterpolator() { }
 
-    public static String interpolate(String line, Properties properties) {
+    public static String interpolate(String line, Map<String, String> context, String filter) {
+        return interpolate(line, context, getExpressionMarkersFromFilter(filter));
+    }
+
+    public static String interpolate(String line, Properties properties, String filter) {
+        return interpolate(line, properties, getExpressionMarkersFromFilter(filter));
+    }
+
+    /**
+     * Replace properties in a string
+     *
+     * @param line              string provided with parameters
+     * @param context           properties provided as hashmap
+     * @param expressionMarkers additional markers to use for parameterized expressions
+     * @return string with properties parameters replaced
+     */
+    public static String interpolate(String line, Map<String, String> context, Map<String, String> expressionMarkers) {
+        Properties p = new Properties();
+        context.forEach(p::put);
+        return checkForPropertiesInLine(line, p, expressionMarkers, null);
+    }
```

Comment:
Unused code, please remove.
If you created either of these unused methods to comply with tests, just update the tests to provide the necessary input parameters for the functions.